enterprise/internal/batches/store/batch_spec_workspaces.go

```diff
@@ -399,6 +399,60 @@ func (s *Store) MarkSkippedBatchSpecWorkspaces(ctx context.Context, batchSpecID
 	return s.Exec(ctx, q)
 }
 
+// ListRetryBatchSpecWorkspacesOpts options to determine which btypes.BatchSpecWorkspace to retrieve for retrying.
+type ListRetryBatchSpecWorkspacesOpts struct {
+	BatchSpecID      int64
+	IncludeCompleted bool
+}
+
+// ListRetryBatchSpecWorkspaces lists all btypes.BatchSpecWorkspace to retry.
+func (s *Store) ListRetryBatchSpecWorkspaces(ctx context.Context, opts ListRetryBatchSpecWorkspacesOpts) (cs []*btypes.BatchSpecWorkspace, err error) {
+	ctx, _, endObservation := s.operations.listRetryBatchSpecWorkspaces.With(ctx, &err, observation.Args{})
+	defer endObservation(1, observation.Args{})
+
+	q := getListRetryBatchSpecWorkspacesQuery(&opts)
+	cs = make([]*btypes.BatchSpecWorkspace, 0)
+	err = s.query(ctx, q, func(sc dbutil.Scanner) error {
+		var c btypes.BatchSpecWorkspace
+		if err := sc.Scan(
+			&c.ID,
+			&jsonIDsSet{Assocs: &c.ChangesetSpecIDs},
+		); err != nil {
+			return err
+		}
+		cs = append(cs, &c)
+		return nil
+	})
+
+	return cs, err
+}
+
+func getListRetryBatchSpecWorkspacesQuery(opts *ListRetryBatchSpecWorkspacesOpts) *sqlf.Query {
+	preds := []*sqlf.Query{
+		sqlf.Sprintf("repo.deleted_at IS NULL"),
+		sqlf.Sprintf("batch_spec_workspaces.batch_spec_id = %s", opts.BatchSpecID),
+	}
+
+	if !opts.IncludeCompleted {
+		preds = append(preds, sqlf.Sprintf("batch_spec_workspace_execution_jobs.state != 'completed'"))
```

Comment:
Total nit, but I'd prefer this to use the Go constant for the value just so it's one less place we have to find in the code if this ever changes:

```suggestion
		preds = append(preds, sqlf.Sprintf("batch_spec_workspace_execution_jobs.state != %s", btypes.BatchSpecWorkspaceExecutionJobStateCompleted))
```

(I think that's the right constant, but you should probably double check.)

Reply:
I agree to use the constant. Totally wasn't thinking - this was copy paste from the SQL function I made